src/main/java/com/transferwise/openbanking/client/api/event/RestEventClient.java

```diff
@@ -184,4 +156,15 @@ private OBErrorResponse1 mapBodyToObErrorResponse(String responseBodyAsString) {
         }
     }
 
+    public static <T> Mono<T> handleWebClientResponseException(WebClientResponseException e, String prefixLog) {
+        var errorMessage = "%s, response status code %s, body returned '%s'".formatted(prefixLog, e.getStatusCode(), e.getResponseBodyAsString());
+        log.error(errorMessage, e);
+        return Mono.error(new EventApiCallException(errorMessage, e));
+    }
+
+    public static <T> Mono<T> handleWebClientException(WebClientException e, String prefixLog) {
+        var errorMessage = "%s, and no response body returned".formatted(prefixLog);
+        log.error(errorMessage, e);
+        return Mono.error(new EventApiCallException(errorMessage, e));
+    }
 }
```

Comment:
We are repeating this functions in each file. Better to move to a util file.

Reply:
Done it for `handleWebClientException`, but we can't really do the same for `handleWebClientResponseException` as some files requires `mapBodyToObErrorResponse` that returns `OBErrorResponse1` of its own.

e.g. `RestPaymentClient` uses `OBErrorResponse1` from `model.payment.OBErrorResponse1` while `RestVrpClient` uses `OBErrorResponse1` from `model.vrp.OBErrorResponse1`